### PR TITLE
Extract SaslClient/SaslServer logic from handlers as abstract classes

### PR DESCRIPTION
**Pull request by ggezer**

### CODE REVIEW

**Comment by ggezer** [comment]
Jenkins, test this please

**Comment by ggezer** [comment]
Jenkins, test this please

**Comment by ggezer** [comment]
@bf8086 PTAL. Thanks!

**Comment by ggezer** [comment]
@bf8086 PTAL.

File: core/common/src/main/java/alluxio/security/authentication/AbstractSaslServerHandler.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.sasl.SaslException;
+import javax.security.sasl.SaslServer;
+
+/**
+ * Abstract {@link SaslServerHandler} implementation that maintains {@link SaslServer} instance.
+ */
+public abstract class AbstractSaslServerHandler implements SaslServerHandler {
```
Comment (bf8086) [inline core/common/src/main/java/alluxio/security/authentication/AbstractSaslServerHandler.java]: These classes looks identical in logic. Is there a way to further unify the code path?

Comment (ggezer) [reply core/common/src/main/java/alluxio/security/authentication/AbstractSaslServerHandler.java]: You mean dropping `SaslServerHandler` maybe?

Comment (bf8086) [reply core/common/src/main/java/alluxio/security/authentication/AbstractSaslServerHandler.java]: not sure about that, I was just thinking of having something like a `AbstractSaslHandler<SaslServer>` so we don't have to repeat the logic in two classes.

Comment (ggezer) [reply core/common/src/main/java/alluxio/security/authentication/AbstractSaslServerHandler.java]: `SaslClient` and `SaslServer` are not sharing an ancestor even though their logic looks identical. That makes it hard to unify their abstract handlers without lots of boilerplate...

**Comment by ggezer** [comment]
@gpang PTAL. Thanks!

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**